File: 136/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        int N = nums.length;
        int [] H = new int[60001];
        for(int i=0; i<N; i++){
            H[nums[i]+30000]++;
        }
        for(int i=0; i<60001; i++){
            if(H[i]%2==1) return i-30000;
        }
        return 0;
    }
}

File: 136/single-number.py
# LeetCode 136. Single Number
# nums 陣列裡「有1個數」只出現1次，找出來 
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0  # XOR 的神奇功能：出現2次=沒有出現
        for num in nums:
            ans ^= num
        return ans
